manually align and reduce memsize internal_state for 64-bit platforms

GermanAizek opened the pull request:
Using pahole memory analyzer from Red Hat Linux. Struct's reserve field can be removed. I could comment field that it is easier for the compiler to work with structures where there is no explicit definition of fields. But I thought I could break alignment in arm and x86.
- reduce 6144 bytes to 6024 bytes because move field reserved for future to arm and x86
- remove aligned keywords

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **Refactor**
  - Simplified struct alignment annotations and adjusted conditional inclusion of padding fields for improved consistency.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Dead2 commented:
I don't see any reason to do this unfortunately.

The aligned hints are there to let the compiler know that the pointers point to aligned buffers, enabling aligned reads/writes where possible.

The reserved bytes are there to let us add/remove parameters without changing the size of the struct, thereby avoiding changes to the size of our alloc calls. Some applications allocate those up-front and expect those not to change (Nginx for example). Additionally the reserved bytes pad the struct out to end on a cacheline boundary.

GermanAizek committed:
manually align and reduce memsize internal_state for 64-bit platforms

- reduce 6144 bytes to 6024 bytes because move field reserved for future to arm and x86
- remove aligned keywords
